pkg/cloudprovider/cloudprovider.go

```diff
@@ -180,6 +181,12 @@ func (c *CloudProvider) Get(ctx context.Context, providerID string) (*karpv1.Nod
 func (c *CloudProvider) GetInstanceTypes(ctx context.Context, nodePool *karpv1.NodePool) ([]*cloudprovider.InstanceType, error) {
 	nodeClass, err := c.resolveNodeClassFromNodePool(ctx, nodePool)
 	if err != nil {
+		// Special case for terminating NodeClass: return empty list instead of error
+		if errors.IsNotFound(err) && strings.Contains(err.Error(), "is terminating") {
+			log.FromContext(ctx).V(4).Info(fmt.Sprintf("NodeClass for NodePool %s is terminating, returning empty instance types list", nodePool.Name))
+			return []*cloudprovider.InstanceType{}, nil
+		}
+
 		if errors.IsNotFound(err) {
 			c.recorder.Publish(cloudproviderevents.NodePoolFailedToResolveNodeClass(nodePool))
 		}
```

Comment:
This seems to directly contradict the guidance on L196 that we should return the info that the NodeClass is terminating back to the caller and handle the error there -- should we just handle this properly by ignoring NotFound inside of the consolidation controller in the upstream code?